app/scripts/controllers/tools.js

```diff
@@ -98,17 +99,53 @@ angular.module('oncokbApp')
                     if ($scope.historySearchResults.length === 0) {
                         $scope.errorMessage = 'Sorry, there are no results that match your search.';
                     } else if ($scope.historySearchResults.length > 0) {
-                        _.each($scope.historySearchResults, function(history) {
-                            _.each(history.records, function(record) {
-                                if (record.old && record.new) {
-                                    record.diffHTML = mainUtils.calculateDiff(record.old, record.new);
-                                }
+                        if ($scope.includeUUID) { // Used for developers to look gene meta data
+                            _.each($scope.historySearchResults, function(history) {
+                                _.each(history.records, function (record) {
+                                    if (record.old && record.new) {
+                                        record.diffHTML = mainUtils.calculateDiff(JSON.stringify(record.old), JSON.stringify(record.new));
+                                    }
+                                });
                             });
-                        });
+                        } else {
+                            _.each($scope.historySearchResults, function(history) {
+                                _.each(history.records, function(record) {
+                                    if (record.operation === 'add') {
```

Comment:
These seem to be new logics added. And after getting the filterHistoryRecord, you don't seem to do anything with it though.

Reply:
oh ok, you just visualize it. But about the  `.old` in the next condition

Reply:
@zhx828 For `add` operation, it only has record.new without record.old. For `delete`, it only has record.old but record.new.